### ISSUE TITLE
Describe triaging process in detail

### ISSUE THREAD

**Issue by mx-psi** [open-telemetry/opentelemetry-collector-contrib#36678]

**Comment by VihasMakwana** [comment]
@mx-psi this makes sense to me.

One concern I have is about the `needs-triage` label.

Should we remove `needs-triage` if the issue is valid or unclear? Or only remove the label if the issue is valid?

**Comment by bacherfl** [comment]
Question regarding the `good-first-issue` tag: should we add an extra step wait for the code owners of the respective component? In some cases an issue might appear straight forward at first, but there may be some unconsidered implications, so it might be better to have deeper insights by the code owners before adding that tag

**Comment by VihasMakwana** [comment]
> should we add an extra step wait for the code owners of the respective component?
+1. Either code owners or one of maintainers.

**Comment by mx-psi** [comment]
Makes sense to me to wait for the codeowners. Maybe the triaging process can include asking the code owners in a comment

**Comment by mx-psi** [comment]
To nudge code owners to fill in certain labels we may want to have things like `priority:unassigned`. We may also want to describe the codeowners workflow (as a separate diagram)

**Comment by crobert-1** [comment]
Another question regarding labels: Do we want to add `waiting-for` labels to issues and PRs instantly, or should we have some amount of delay? Historically I've left issues for a few days before adding the label to try to avoid coming across too strongly to contributors, but maybe we shouldn't worry about that?

Argument against a delay: Adding these labels right away will give us instant visibility into the current state of issues and PRs. It will save triagers time in looking through open issues, as they can simply see there's nothing that can be done for an issue at the present time.

Argument for a delay: Instantly adding these labels may come across as a bit aggressive to contributors if we don't communicate their purpose clearly. If I as a code owner see a new issue was opened an hour ago, it may be off-putting if there's already a label waiting for me. Another concern is out of date labels (this is a concern either way, but more so if we're applying the label more often). If a code owner and issue author are actively discussing an issue, the label would need to be switched back and forth, or else it would go stale. If someone is only glancing at labels to know if an issue needs attention, they would be misled by the labels. Also, you're not subscribed to issues when adding labels, so the original triager would not be notified if the labels were rendered outdated.

There was some discussion in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/36679 and we may end up assigning issues to triagers, which would be helpful for my concern around outdated labels.

I don't think it's too big of a deal either way, the key is to communicate clearly the purpose of these labels to ensure they're received well.

**Comment by VihasMakwana** [comment]
@mx-psi I can work on this. I'll add a flowchart similar to what I did for https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/36677.

**Comment by VihasMakwana** [comment]
PR is up for review: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/40805
